File: backend/src/main/java/com/grouply/backend/chat_message/dto/ChatMessageDTO.java
package com.grouply.backend.chat_message.dto;

import com.grouply.backend.chat_member.dto.ChatMemberDTO;
import com.grouply.backend.chat_message.ChatMessageStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatMessageDTO {
    private Long id;
    private String content;
    private ChatMessageStatus status;
    private Instant sentAt;
    private ChatMemberDTO sender;
    private Long chatRoomId;
}

File: frontend/src/components/layout/message-box/model/chatmessageDTO.ts
import type { ChatMessageStatus } from "./chat.message.status";
import type { ChatMemberDTO } from "./chatmember.DTO";


export interface ChatMessageDTO {
  id: number;
  content: string;
  status: ChatMessageStatus;
  sentAt: string;        
  sender: ChatMemberDTO;
  chatRoomId: number;
}